lib/improver/weights.py

```diff
@@ -451,3 +451,110 @@ def __repr__(self):
         desc = ('<ChooseDefaultWeightsNonLinear '
                 'cval={0:4.1f}>'.format(self.cval))
         return desc
+
+
+class ChooseDefaultWeightsTriangular(object):
+    """ Calculate Default Weights using a Triangular Function. """
+    def __init__(self, width, midpoint, units="no_unit"):
+        """Set up for calculating default weights using triangular function.
+
+            Args:
+                width : float, the width of the triangular function from the
+                        centre point.
+                midpoint : float, the centre point of the triangular function.
+                units : cf_units.Unit, the cf units of the width and midpoint.
+        """
+        self.width = width
+        self.midpoint = midpoint
+        if not isinstance(units, cf_units.Unit):
+            units = cf_units.Unit(units)
+        self.parameters_units = units
+
+    def triangular_weights(self, coord_vals):
+        """Create triangular weights.
+
+            Args:
+                coord_vals : numpy array
+                             An array of coordinate values that we want to
+                             caluculate weights for.
+
+            Returns:
+                weights : array of weights, sum of all weights = 1.0
+
+            Raises:
```

Comment:
I don't think a `Raises` section is still required, if nothing's raised. Currently, there's just a `.` in this section.